spec/spec_helper.rb

```diff
@@ -70,6 +69,19 @@ def hash_inspect(hash)
 end
 RSpec.configuration.include_context "with warn_about_potential_false_positives set to false", :warn_about_potential_false_positives
 
+RSpec.shared_context "with modified configuration" do
+  around do |example|
+    configuration = example.metadata[:with_configuration]
+    key = configuration.keys.first
+    value = configuration.values.first
```

Comment:
I mean if you're going to extract this, you might as well detect / presume the hash and map on them...

Reply:
Sure, I've added to my todo to do that, and also to rework "with warn_about_potential_false_positives" and other shared examples that flip configuration bits.
Might even be extracted to Support if it's useful in other repos.

But I decided to leave this refactoring for later, to after 4.0.

Reply:
Please do it here first. Theres no point this being a hash and not used.

Reply:
> Theres no point this being a hash and not used

I tend to agree and started implementing it this way, but it looked quite clumsy (due to the lack of `Hash#slice` in Ruby 2.3).
Another approach could be to use two different metadata keys, `with_configuration_key` and `with_configuration_value`, quite clumsy too.

The value passed to `with_configuration` could be an array and using `first`/`last`. That imposes the question of how it should behave if we would be passing an array with more than two entries.

So, until this shared example gained some critical mass of usages of it, I decided not to grow its complexity.

Reply:
> So, until this shared example gained some critical mass of usages of it, I decided not to grow its complexity.

It currently has unnecessary complexity for its use case in the form of the dynamic methods anyway, either simplify it so its specific to the config we're changing, or complete it please :)

Reply:
Done!

Reply:
I can also rework existing usages of alternative approaches but would prefer to do this later, after https://github.com/rspec/rspec-expectations/pull/1253 is merged and after we remove `include_chain_clauses_in_custom_matcher_descriptions` et al to avoid unnecessary work.